### PR TITLE
Complete all README info

### PR DESCRIPTION
**Pull request by antlet**
Adds all the information to the readme file that helps the user with installing and using GridGopher.

### CODE REVIEW

**Comment by noorbuchi** [comment]
I think this PR is ready for review and merge. There are still two sections of the README that are left unfinished but considering that we're still developing the tool, they would likely need to be revised a lot anyways. Let's get this checked for clarity, typos, and grammar mistakes and proceed with the merge if everything looks good.

File: README.md [hunk 1]
```diff
@@ -1,13 +1,81 @@
...
+[![codecov](https://codecov.io/gh/noorbuchi/GridGopher/branch/main/graph/badge.svg?token=02353FAN4W)](https://codecov.io/gh/noorbuchi/GridGopher)
+![stars](https://img.shields.io/github/stars/noorbuchi/GridGopher.svg)
 
-## Important Notes
+![GridGopherLogo](images/GopherLogo.png)
 
-- Values in the `.env` file must be surrounded by double quotation marks `"`
-  otherwise, newline character `\n` will cause issues.
+GridGopher is a plugin friendly tool that allows users to connect collected data
+from Google Sheets and GitHub issue trackers and pull requests. The tool
```
Comment (antlet) [inline README.md:12]: I'm not sure if the use of multiple and's is grammatically correct. Feel free to disregard this comment depending on what you think/research.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,81 @@
...
+provides the basic API and encourages users to utilize it in their applications.
 
-Expected Format:
+## Set Up and Installation
+
+**Installation using `pip` is currently not supported. However, it's planned for
```
Comment (antlet) [inline README.md:17]: Consider using "it is" instead of "it's". Usually it is more formal that way.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,81 @@
...
+future releases**
+
+GridGopher uses Poetry to create a Python virtual environment and manage
+dependencies. For more information about Poetry, check out [the
+documentation](https://python-poetry.org/). To set up the tool for use, please
+follow the steps outlined below:
+
+**1- Install Poetry:**
+
+Install poetry using the steps outlined
```
Comment (antlet) [inline README.md:27]: I noticed that Poetry is usually uppercase, but it is not in this instance.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,81 @@
...
+[here](https://python-poetry.org/docs/#installation). To verify that poetry was
```
Comment (antlet) [inline README.md:28]: Same Poetry comment as above.

File: README.md [hunk 5]
```diff
@@ -1,13 +1,81 @@
...
+
+```
+poetry -V
+```
+
+The expected output is the version of Poetry installed.
+
+**2- Install Python Dependencies:**
+
+Once poetry has been installed successfully, clone or download the repository
```
Comment (antlet) [inline README.md:39]: Same Poetry comment as above.

File: README.md [hunk 6]
```diff
@@ -1,13 +1,81 @@
...
+and navigate to the root of the repository. Use the following command to install
+all the dependencies used by GridGopher:
+
+```
+poetry install
+```
+
+This command might take some time to finish running. Once it's completed,
```
Comment (antlet) [inline README.md:47]: Refer to previous comment about "it's".

File: README.md [hunk 7]
```diff
@@ -1,13 +1,81 @@
...
+GridGopher is ready for use!
+
+## Running GridGopher
+
+### API Setup
+
+GridGopher requires authentication tokens for a Google Sheets API service
```
Comment (antlet) [inline README.md:54]: Check the overall grammar of this sentence. It seems like the 's' on "accounts" should be deleted, but I am not entirely sure.

File: README.md [hunk 8]
```diff
@@ -19,3 +87,58 @@ TOKEN_URI="value"
 AUTH_PROVIDER_X509_CERT_URL="value"
 CLIENT_X509_CERT_URL="value"
 ```
+
+#### Github Access Token
+
+If you intend to use GridGopher's GitHub interactions features, it's required to
```
Comment (antlet) [inline README.md:93]: Refer to previous comment about "it's".

File: README.md [hunk 9]
```diff
@@ -1,13 +1,81 @@
...
+up a service account and get the tokens, please follow the steps below:
+
+#### Google Sheets Service Account
+
+[This tutorial](https://youtu.be/4ssigWmExak?t=215)
+from `3:35` until `8:20` gives clear and
+detailed steps on how to create a service account and create an authentication
+key. However, it includes extra steps that not everyone will need to follow. You
+can follow the video if preferred or the [Sheets API
+Guide](docs/Google_API_Setup.md)
```
Comment (antlet) [inline README.md:65]: Add a period after the link and parenthesis.

File: README.md [hunk 10]
```diff
@@ -19,3 +87,58 @@ TOKEN_URI="value"
...
+### Writing Config
+
+GridGopher relies on user written YAML configuration to collect data from Google
+Sheets and organize it in regions. GitHub interactions are also managed by
+YAML configuration. To read more about the structure of GridGopher
+configuration, please refer to our [schema documentation](docs/schemas.md).
+
+#### Sheets Collector
+
+Sheets Collector the component of GridGopher responsible for making Google
```
Comment (antlet) [inline README.md:126]: Change sentence to 'Sheets Collector is the component of GridGopher that is responsible for making Google...' or something similar. The main missing element is the word "is".

File: README.md [hunk 11]
```diff
@@ -19,3 +87,58 @@ TOKEN_URI="value"
...
+Sheets API calls and retreiving data from user specified files and sheets.
+Additionally, it creates an object oriented structure for regions and sheets
+of data. In order to use this component, configuration YAML files are needed
+in the `config/sheet_sources` directory. Multiple files can be used if multiple
+sheets are being read at the same time.
+
+#### Github Interactions
+
+Another component of GridGopher is the GitHub Interaction interface. It's
```
Comment (antlet) [inline README.md:135]: Refer to previous comment about "it's".

File: README.md [hunk 12]
```diff
@@ -19,3 +87,58 @@ TOKEN_URI="value"
...
+responsible for making API requests to GitHub and posting user specified
+information to GitHub in the form of Issue Trackers, Pull Requests, and files.
+The user has complete control of this component's behavior through YAML
+configurations files found in `config/github_interactions` directory. Multliple
```
Comment (antlet) [inline README.md:139]: Remove 's' from 'configurations'.

File: README.md (hunk 1 above)
Comment (antlet) [inline README.md:12]: I noticed that the terms 'issue trackers' and 'pull requests' were used multiple times in the readme but have different cases throughout. Consider capitalizing the terms to keep things uniform.

File: docs/schemas.md [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+        },
+    },
+}
+```
+
+## GitHub Interactions Schema
+
+This type of schema describes the structure of configurations found in the
+`config/github_interactions` directory. This configuration is used to post
+entries to Github such as issue trackers, pull requests, and files.
```
Comment (antlet) [inline docs/schemas.md:177]: Refer to previous comment about case-uniformity with these two terms.

File: README.md [hunk 14]
```diff
@@ -19,3 +87,58 @@ TOKEN_URI="value"
...
+Additionally, it creates an object oriented structure for regions and sheets
+of data. In order to use this component, configuration YAML files are needed
+in the `config/sheet_sources` directory. Multiple files can be used if multiple
+sheets are being read at the same time.
+
+#### Github Interactions
+
+Another component of GridGopher is the GitHub Interaction interface. It's
+responsible for making API requests to GitHub and posting user specified
+information to GitHub in the form of Issue Trackers, Pull Requests, and files.
```
Comment (antlet) [inline README.md:137]: Here the two terms are capitalized. Change or ignore this based on what it decided for uniformity.

File: docs/Google_API_Setup.md [hunk 15]
```diff
@@ -1,21 +1,49 @@
...
+# Google API Setup
+
+- Create a Google API project from [cloud console](https://console.cloud.google.com/)
+  - Click on select project, or create new project
+
+![new_project1](../images/new_project1.png)
+![new_project2](../images/new_project2.png)
+![new_project3](../images/new_project3.png)
+
+- Enable Google sheets API from the [API Menu](https://console.cloud.google.com/apis/dashboard)
```
Comment (antlet) [inline docs/Google_API_Setup.md:10]: Capitalize 'sheets' to keep uniformity with other instances.

File: docs/Google_API_Setup.md [hunk 16]
```diff
@@ -1,21 +1,49 @@
...
+
+- This information can be left blank
+
+![credentials6](../images/credentials6.png)
+
+- Click on the newly created service account and navigate to the keys tab
+
+![credentials7](../images/credentials7.png)
+
+- click on `Add Key` and select `Create new key`
```
Comment (antlet) [inline docs/Google_API_Setup.md:42]: Capitalize 'click'

File: docs/schemas.md [hunk 17]
```diff
@@ -0,0 +1,183 @@
+# Config Schemas
+
+The examples shown here demonstrate the schemas for configuration read by
+GridGopher. In order to for the tool to function with no errors, user written
```
Comment (antlet) [inline docs/schemas.md:4]: Delete 'to'

File: docs/schemas.md [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+Sheets schema describe the format for configuration found in the
+`config/sheet_sources` directory, which are used to retrieve Google Sheet data.
+
+### Defining Objects
+
+There are two main nested object structures used in the Sheets schema.
+
+#### Region Object
+
+This is the simples object that does not contain complex nested object in it. It
```
Comment (antlet) [inline docs/schemas.md:18]: I believe 'simples' was meant to be typed as 'simplest'.

**Review by antlet — COMMENTED** [review]
I added many comments about small grammatical concerns that I noticed. It is very nit-picky, but I figured that they can be fixed now rather than later. Once they are all addressed, please leave a new comment saying so and I will re-review and approve.

**Comment by noorbuchi** [comment]
Thank you @antlet for the review. I pushed all the changes that your requested. I also added table of contents for the README and Schemas markdown file to make them easier to navigate. Please take another look at the changes when you get a chance. Thanks!

File: README.md [hunk 19]
```diff
@@ -1,13 +1,95 @@
 # GridGopher
 
-A placeholder repository for Allegheny College automated grade transfer tool
+![License](https://img.shields.io/badge/license-MIT-blue?style=flat)
+![BuiltWith](https://img.shields.io/badge/Built%20With-Python-blue?style=flat&logo=python&logoColor=yellow)
+![Actions Status](https://github.com/noorbuchi/GridGopher/workflows/Lint%20and%20Test/badge.svg)
+[![codecov](https://codecov.io/gh/noorbuchi/GridGopher/branch/main/graph/badge.svg?token=02353FAN4W)](https://codecov.io/gh/noorbuchi/GridGopher)
+![stars](https://img.shields.io/github/stars/noorbuchi/GridGopher.svg)
```
Comment (noorbuchi) [inline README.md:8]: sdfsa

File: README.md [hunk 20]
```diff
@@ -1,13 +1,95 @@
...
+
+**Installation using `pip` is currently not supported. However, it is planned for
+future releases**
+
+GridGopher uses Poetry to create a Python virtual environment and manage
+dependencies. For more information about Poetry, check out [the
+documentation](https://python-poetry.org/). To set up the tool for use, please
+follow the steps outlined below:
+
+**1- Install Poetry:**
```
Comment (noorbuchi) [inline README.md:39]: dfsdf

**Review by antlet — COMMENTED** [review]
I am the initial author of this PR so I cannot approve changes, but I unofficially approve the changes that were made.

**Review by noorbuchi — APPROVED** [review]
Good to merge!

**Review by tuduun — APPROVED** [review]

**Pull request merged**